src/nft/components/view/NFTLastSoldPrice.tsx

```diff
@@ -12,14 +12,18 @@ export function NFTLastSoldPrice({
   className,
   label = 'Mint price',
 }: NFTLastSoldPriceReact) {
-  const {
-    lastSoldPrice: { amount, currency, amountUSD },
-  } = useNFTContext();
+  const { lastSoldPrice } = useNFTContext();
 
-  if (!amount || !currency || !amountUSD) {
+  if (
+    !lastSoldPrice?.amount ||
+    !lastSoldPrice?.currency ||
+    !lastSoldPrice?.amountUSD
+  ) {
     return null;
   }
 
+  const { amount, currency, amountUSD } = lastSoldPrice;
```

Comment:
I think restructuring should still work here if the values are undefined?

```
const { amount, currency, amountUSD } = useNFTContext();
if (!amount || !currency || !amountUSD) {
  return null;
}
```

Reply:
only if i default lastSoldPrice to {}

Reply:
`  const {
    lastSoldPrice: { amount, currency, amountUSD } = {},
  } = useNFTContext();
` I always feel those destructure defaults are kind of messy

Reply:
oo yeah I don't like that